spell_check.py

```diff
@@ -0,0 +1,88 @@
+#!/bin/python
+
+import subprocess
+import os
+import argparse
+import yaml
+from spellchecker import SpellChecker
+
+spell = SpellChecker()
+
+def spelling(text):
+    words = spell.split_words(text)
+    misspelled = spell.unknown(words)
+
+    misspelled = [x for x in misspelled if x.strip() and x]
+    return misspelled
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Process different file types with specific options")
+    parser.add_argument("-t", "--type", choices=["txt", "yaml"], required=True, help="File type (currently supports 'txt', 'yaml')")
+    parser.add_argument("-k", "--keys", nargs="+", help="List of YAML fields (only applicable for YAML files)")
+    parser.add_argument("-f", "--files", nargs="+", required=True, help="List of input file paths")
+    parser.add_argument("-d", "--dictionaries", nargs="+", help="List of dictionary files")
+
+    args = parser.parse_args()
+    return args
+
+def process_yaml_file(file_path, keys):
+    file = open(file_path, "r")
+
+    data = yaml.safe_load(file)
+    for key in keys:
+        if not key in data:
+            exit(f"key {key} not found")
+
+        misspellings = spelling(data[key])
+        if len(misspellings) > 0:
+            print(f"Misspellings in {file_path} in \"{key}\":")
+            for m in misspellings:
+                print(f"\t{m}")
+
+def process_yaml_files(file_paths, keys=None):
+    if keys:
+        for file_path in file_paths:
+            process_yaml_file(file_path, keys)
+    else:
+        process_txt_files(file_paths)
+
+def process_txt_file(file_path):
+    file = open(file_path, "r")
+    data = file.read()
+    misspellings = spelling(data)
+    if len(misspellings) > 0:
+        print(f"Misspellings in {file_path}:")
+        for m in misspellings:
+            print(f"\t{m}")
+
+def process_txt_files(file_paths):
+    for file_path in file_paths:
+        process_txt_file(file_path)
+
+def main():
+    args = parse_args()
+    dictionaries = ["./dictionary.txt"]
+
+    if args.dictionaries != None:
+        dictionaries = args.dictionaries
+    else:
+        if not os.path.exists(dictionaries[0]):
+            print(f"Default dictionary {dictionaries[0]} does not exist.")
+            print(f"Consider using the ci provided dictionary using \"-d ci/dictionary.txt\"")
+            exit(-1)
+
+    for d in dictionaries:
+        spell.word_frequency.load_text_file(d)
+
+    if args.type == "txt":
+            process_txt_files(args.files)
+    if args.type == "yaml":
```

Comment:
Shouldn't be a `else` here and the last branch removed since the `parse_args` already validated the `args.type` variable?

Reply:
Ok